_posts/2017-04-30-c.md

```diff
@@ -4,8 +4,15 @@ code: |-
   #include <stdio.h>
 
   int main(int argc, char** argv) {
+    printf("%f\n", .1 + .2); // defaults to 6-digit precision
     printf("%.17f\n", .1 + .2);
+    printf("%g\n", .1 + .2); // defaults to significant precision
```

Comment:
What do you mean "significant" precision?

Reply:
The difference between these two outputs.

0.30000000000000004
0.3
If you change 0.2 to 0.21 you get
0.31